app/services/agreements/iaa_order_seeder.rb

```diff
@@ -0,0 +1,48 @@
+module Agreements
+  class IaaOrderSeeder < BaseSeeder
+    private
+
+    def record_class
+      IaaOrder
+    end
+
+    def filename
+      'iaa_orders.yml'
+    end
+
+    def primary_attribute_bundle(config)
+      { 'iaa_gtc_id' => config['iaa_gtc'].id, 'order_number' => config['order_number'] }
+    end
+
+    def process_config(iaa_name, config)
+      match = /^(?<gtc_number>[A-Z0-9]+)-(?<order_number>\d+)$/.match(iaa_name)
+      config['iaa_gtc'] = IaaGtc.find_by!(gtc_number: match[:gtc_number])
+      config['iaa_status'] =
+        IaaStatus.find_by!(name: config['iaa_status'])
+
+      # this is used in the after_seed method to generate the associated
+      # IntegrationUsages after the IaaOrders are created / updated.
+      @associated_integrations ||= {}
+      key = [config['iaa_gtc'].id, match[:order_number]]
+      @associated_integrations[key] = config['integrations']
+
+      permitted_attrs =
+        %w[mod_number start_date end_date estimated_amount iaa_status iaa_gtc]
+      config.slice(*permitted_attrs).merge('order_number' => match[:order_number])
+    end
+
+    def after_seed
+      @associated_integrations.each do |order_info, integrations|
+        next if integrations.blank?
+
+        order = IaaOrder.find_by!(iaa_gtc_id: order_info[0], order_number: order_info[1])
```

Comment:
we can destructure arrays so we don't have to do `[0]` and `[1]`:

```suggestion
      @associated_integrations.each do |(iaa_gtc_id, order_number), integrations|
        next if integrations.blank?

        order = IaaOrder.find_by!(iaa_gtc_id: iaa_gtc_id, order_number: order_number)
```

Reply:
Great suggestion, it definitely smelled funny to me. I'll update